### PR TITLE
Restore old cl kernel v4

### PR DESCRIPTION
**Pull request by EoD**
This change restores the old kernel and adds the new kernel as an "unstable" option (see ``--kernel`` parameter).

### CODE REVIEW

**Comment by chfast** [comment]
Firstly I was thinking that we should should merge both kernel and find the regression, but having multiple kernels might be good idea - we can get more benchmarks from users.

**Comment by EoD** [comment]
@goobur How do we call the two kernels? I currently call them "Original" and "SGMiner", but the latter one is only due to the lack of understanding of waht really changed.

**Comment by chfast** [comment]
We can call them "default" or "stable" and "experimental".

**Comment by EoD** [comment]
@chfast Don't you think that we might need to have multiple stable kernels in the future? For instance, optimized for different hardware (generations).
I am going for "stable", "unstable", "experimental" until we are able to answer the question.

**Comment by EoD** [comment]
Squashed some commits and reworded the commit message. This PR is done from my point of view.

File: ethminer/MinerAux.h [hunk 1]
```diff
@@ -314,6 +314,18 @@ class MinerCLI
 			}
 		else if (arg == "--list-devices")
 			m_shouldListDevices = true;
+		else if ((arg == "-k" || arg == "--kernel") && i + 1 < argc)
+		{
```
Comment (smurfy) [inline ethminer/MinerAux.h]: since this is OpenCL Specifiy i would call it somthing like the other ocl params.

File: ethminer/MinerAux.h [hunk 2]
```diff
@@ -1047,6 +1066,7 @@ class MinerCLI
 	unsigned m_openclPlatform = 0;
 	unsigned m_miningThreads = UINT_MAX;
 	bool m_shouldListDevices = false;
+	unsigned m_selectedKernel = 0;  ///< A generic value containing a kernel enum. Different between OCL and CUDA.
```
Comment (smurfy) [inline ethminer/MinerAux.h]: As said before would not use the same variable for ocl + cuda. since you can use ``-X`` for both if you have mixed cards.

Comment (EoD) [reply ethminer/MinerAux.h]: Fair enough, I didn't think of mixed systems.

**Comment by EoD** [comment]
I added @smurfy's requested changes and added one more commit which expands the help page to clearly distinguish OCL-only/CUDA-only parameters.

File: ethminer/MinerAux.h (hunk 2 above)
Comment (smurfy) [reply ethminer/MinerAux.h]: yes, that would be premium :)

**Comment by chfast** [comment]
We use too many global values for config. They should be moved to Miner classes.

**Pull request merged**